conformance/lang/expressions/logical-expressions/logical_and.balt

```diff
@@ -0,0 +1,732 @@
+    boolean _ =  check int:fromString("123") && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using trap expression.
+Labels: logical-expr, trap-expr
+
+function init() {
+    boolean _ =  true && trap divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  divideNum(1, 0) && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  divideNum(1, 0) && divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+}
+
+function divideNum(int a, int b) returns int {
+    return a / b;
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using xml navigate expression.
+Labels: logical-expr, xml-navigate-expr
+
+function init() {
+    xml a = xml `<root><ele>true</ele><ele>false</ele></root>`;
+    boolean _ =  true && a/*; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/* && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/* && a/*; // @error static type of each expression must be a subtype of boolean
+    
+    boolean _ =  true && a/<ele>; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/<ele> && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/<ele> && a/<ele>; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ =  true && a/<ele>/*; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/<ele>/* && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a/<ele>/* && a/<ele>/*; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: output
+Description: Test logical expression output.
+Labels: logical-expr
+
+function init() {
+    io:println(true && true); // @output true
+    io:println(false && true); // @output false
+    io:println(true && false); // @output false
+    io:println(false&& false); // @output false
+}
+
+Test-Case: output
+Description: Test logical expression output type.
+Labels: logical-expr, is-expr
+
+function init() {
+    io:println(true && true is boolean); // @output true
+    io:println(false && true is boolean); // @output true
+    io:println(true && false is boolean); // @output true
+    io:println(false && false is boolean); // @output true
+}
+
+Test-Case: output
+Description: Test static type of expression in logical expression using literal unions.
+Fail-issue: ballerina-lang #33359
+Labels: logical-expr, union-type
+
+function init() {
+    boolean|boolean a = true; // @output true
```

Comment:
Don't we have tests assigning to just `boolean`?

How do we test the static type? We need negative tests also.

Reply:
It is tested via invalid types with `Test-Case: error` type tests